### PR TITLE
get_var_integer update

### PR DESCRIPTION
**Pull request by OdyWayne**

### CODE REVIEW

File: src/mqtt/transport/tcp/mqtt_tcp.c [hunk 1]
```diff
@@ -803,7 +802,10 @@ mqtt_tcptran_pipe_send_start(mqtt_tcptran_pipe *p)
 		niov++;
 	}
 	// assure send correct packet
-	len = get_var_integer((header + 1), &len_of_var);
+	int ret;
+	NNI_ASSERT((ret = mqtt_get_remaining_length(header,
+	                (uint32_t) nni_msg_header_len(msg), &len, NULL)) ==
+	    MQTT_SUCCESS);
```
Comment (wanghaEMQ) [inline src/mqtt/transport/tcp/mqtt_tcp.c:808]: Add handler for release mode when remaining length is not equal to nng_msg_len

File: src/sp/transport/mqtt/broker_tcp.c [hunk 2]
```diff
@@ -1067,12 +1067,14 @@ nmq_pipe_send_start_v4(tcptran_pipe *p, nni_msg *msg, nni_aio *aio)
...
-				property_len = get_var_integer(
-				    body + 4 + tlen, &property_bytes);
+				mqtt_get_remaining_length(body + 4 + tlen - 1,
+				    (uint32_t) nni_msg_header_len(msg),
+				    &property_len, &property_bytes);
 
 			} else {
-				property_len = get_var_integer(
-				    body + 2 + tlen, &property_bytes);
+				mqtt_get_remaining_length(body + 2 + tlen - 1,
```
Comment (wanghaEMQ) [inline src/sp/transport/mqtt/broker_tcp.c:1075]: NO!!! The function name is not quite proper.

Comment (wanghaEMQ) [reply src/sp/transport/mqtt/broker_tcp.c:1075]: The function is not conformed to its usage

File: src/sp/protocol/mqtt/nmq_mqtt.c [hunk 3]
```diff
@@ -1084,7 +1084,7 @@ nano_pipe_recv_cb(void *arg)
 		nni_mtx_unlock(&p->lk);
 
 		if (cparam->pro_ver == MQTT_PROTOCOL_VERSION_v5) {
-			len = get_var_integer(ptr + 2, &len_of_varint);
+			mqtt_get_remaining_length(ptr + 1, (uint32_t) nni_msg_header_len(msg), &len, &len_of_varint);
```
Comment (wanghaEMQ) [inline src/sp/protocol/mqtt/nmq_mqtt.c:1087]: Why no NNI_ASSERT here?

File: src/sp/protocol/mqtt/nmq_mqtt.c [hunk 4]
```diff
@@ -1110,7 +1110,7 @@ nano_pipe_recv_cb(void *arg)
...
```
Comment (wanghaEMQ) [inline src/sp/protocol/mqtt/nmq_mqtt.c:1113]: Why no NNI_ASSERT here?

File: src/sp/transport/mqtt/broker_tcp.c [hunk 5]
```diff
@@ -1117,8 +1119,12 @@ nmq_pipe_send_start_v4(tcptran_pipe *p, nni_msg *msg, nni_aio *aio)
 			}
 		}
 		// copy remaining length
+		uint32_t remaining_length;
+		mqtt_get_remaining_length(header,
+		    (uint32_t) nni_msg_header_len(msg), &remaining_length,
+		    &pos);
```
Comment (wanghaEMQ) [inline src/sp/transport/mqtt/broker_tcp.c:1125]: Why no NNI_ASSERT here?

**Review by wanghaEMQ — CHANGES_REQUESTED** [review]

File: src/sp/transport/mqtt/broker_tcp.c (hunk 2 above)
Comment (OdyWayne) [reply src/sp/transport/mqtt/broker_tcp.c:1075]: `get_remaining_length` can better illustrate its usage than `get_var_integer`, isn't it?

Comment (wanghaEMQ) [reply src/sp/transport/mqtt/broker_tcp.c:1075]: Especially, it has a prefix `mqtt_`

File: src/mqtt/transport/tcp/mqtt_tcp.c (hunk 1 above)
Comment (wanghaEMQ) [reply src/mqtt/transport/tcp/mqtt_tcp.c:808]: This must be done

**Review by wanghaEMQ — CHANGES_REQUESTED** [review]

**Comment by OdyWayne** [comment]
> just found out read_variable_integer is a better alternative

`read_variable_integer` uses `struct pos_buf` which is only used for supplementary apis in `parser` or `code_c`. Better not directly use in trans layer. Is it better to package a new api with `read_variable_integer` to replace `get_var_integer`? @JaylinYu

**Pull request closed without merging**